hub-detect/src/main/resources/hub-detect-4-sh

```diff
@@ -0,0 +1,104 @@
+  if [[ $i == --blackduck.hub.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.hub.password=<redacted>"
+  elif [[ $i == --blackduck.hub.proxy.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.hub.proxy.password=<redacted>"
+  elif [[ $i == --blackduck.hub.api.token=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.hub.api.token=<redacted>"
+  elif [[ $i == --blackduck.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.password=<redacted>"
+  elif [[ $i == --blackduck.proxy.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.proxy.password=<redacted>"
+  elif [[ $i == --blackduck.api.token=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.api.token=<redacted>"
+  else
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS $i"
+  fi
+done
+
+run() {
+  get_detect
+  run_detect
+}
+
+get_detect() {
+  if [ -z "${DETECT_SOURCE}" ]; then
+    if [ -z "${DETECT_RELEASE_VERSION}" ]; then
+      VERSION_CURL_CMD="curl ${DETECT_CURL_OPTS} --silent --header \"X-Result-Detail: info\" 'https://test-repo.blackducksoftware.com/artifactory/api/storage/bds-integrations-release/com/blackducksoftware/integration/hub-detect?properties=${DETECT_VERSION_KEY}' | grep \"${DETECT_VERSION_KEY}\" | sed 's/[^[]*[^\"]*\"\([^\"]*\).*/\1/'"
+      DETECT_SOURCE=$(eval $VERSION_CURL_CMD)
+    else
+      DETECT_SOURCE="https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/${DETECT_RELEASE_VERSION}/hub-detect-${DETECT_RELEASE_VERSION}.jar"
+    fi
+  fi
+  echo "will look for : ${DETECT_SOURCE}"
+
+  DETECT_FILENAME=${DETECT_FILENAME:-$(awk -F "/" '{print $NF}' <<< $DETECT_SOURCE)}
+  DETECT_DESTINATION="${DETECT_JAR_PATH}/${DETECT_FILENAME}"
+
+  USE_REMOTE=1
+  if [ ! -f $DETECT_DESTINATION ]; then
+    echo "You don't have the current file, so it will be downloaded."
+  else
+    echo "You have already downloaded the latest file, so the local file will be used."
+    USE_REMOTE=0
+  fi
+
+  if [ $USE_REMOTE -eq 1 ]; then
+    echo "getting ${DETECT_SOURCE} from remote"
+    curl $DETECT_CURL_OPTS --silent -L -o $DETECT_DESTINATION "${DETECT_SOURCE}"
+    echo "saved ${DETECT_SOURCE} to ${DETECT_DESTINATION}"
+  fi
+}
+
+run_detect() {
+  JAVACMD="java ${DETECT_JAVA_OPTS} -jar ${DETECT_DESTINATION}"
+  echo "running detect: ${JAVACMD} ${LOGGABLE_SCRIPT_ARGS}"
+
+  # first, silently delete (-f ignores missing
+  # files) any existing shell script, then create
+  # the one we will run
+  rm -f $DETECT_JAR_PATH/hub-detect-java.sh
+  echo "#!/bin/sh" > $DETECT_JAR_PATH/hub-detect-java.sh
+  echo "" >> $DETECT_JAR_PATH/hub-detect-java.sh
+  echo $JAVACMD $SCRIPT_ARGS >> $DETECT_JAR_PATH/hub-detect-java.sh
+  source $DETECT_JAR_PATH/hub-detect-java.sh
+  RESULT=$?
+  echo "Result code of ${RESULT}, exiting"
+  rm -f $DETECT_JAR_PATH/hub-detect-java.sh
```

Comment:
Why remove the generated script? If an error is encountered, couldn't the script help with troubleshooting the issue?

Reply:
Excellent point. On it.